### PR TITLE
Feature/tournaments

### PR DESCRIPTION
**Pull request by WojciechAdamiec**

### CODE REVIEW

**Review by Michal-Miko — APPROVED** [review]

File: backend/app/main.py [hunk 1]
```diff
@@ -167,6 +167,9 @@ def update_player(
     if access:
         if crud.get_player(db, player_id=player_id) is None:
             raise HTTPException(status_code=404, detail="Player not found")
+        player_check = crud.get_player_by_name(db, name=player.name)
+        if player_check is not None:
+            raise HTTPException(status_code=404, detail="Name already used")
```
Comment (bakowroc) [inline backend/app/main.py:172]: Should be more like `400` instead `404`

File: backend/app/main.py [hunk 2]
```diff
@@ -375,6 +378,25 @@ def read_matches(
...
+):
+    access = permissions.is_accessible(
+        db=db, firebase_id=firebase_id, clearance="player"
+    )
+    if access:
+        matches = crud.get_upcoming_matches(db, skip=skip, limit=limit)
+        return matches
+    else:
+        raise HTTPException(
+            status_code=404, detail="Permission denied, requires at least: player"
```
Comment (bakowroc) [inline backend/app/main.py:396]: `401` / `403` ?

File: backend/app/main.py [hunk 3]
```diff
@@ -391,11 +413,10 @@ def read_match(
         return db_match
     else:
         raise HTTPException(
-            status_code=404, detail="Permission denied, requires at least: player"
-        )
+            status_code=404, detail="Permission denied, requires at least: player")
```
Comment (bakowroc) [inline backend/app/main.py:416]: Same here  `401` / `403`

File: backend/app/main.py [hunk 4]
```diff
@@ -407,10 +428,184 @@ def update_match(
...
+            check = crud.get_team(db, team_id)
+            if check is None:
+                raise HTTPException(
+                    status_code=404, detail="Team " + str(team_id) + " not found"
+                )
+
+        return crud.create_tournament(db=db, tournament=tournament)
+    else:
+        raise HTTPException(
+            status_code=404, detail="Permission denied, requires at least: moderator"
```
Comment (bakowroc) [inline backend/app/main.py:462]: Same here `401` / `403`

File: backend/app/main.py [hunk 5]
```diff
@@ -407,10 +428,184 @@ def update_match(
...
+    tournament: schemas.TournamentCreate,
+    firebase_id: str = Header(None),
+    db: Session = Depends(get_db),
+):
+    access = permissions.is_accessible(
+        db=db, firebase_id=firebase_id, clearance="moderator"
+    )
+    if access:
+        if tournament.tournament_type not in ["round-robin"]:
+            raise HTTPException(status_code=404, detail="Tournament type unknown")
```
Comment (bakowroc) [inline backend/app/main.py:450]: `400` ?

File: backend/app/main.py [hunk 6]
```diff
@@ -407,10 +428,184 @@ def update_match(
...
+):
+    access = permissions.is_accessible(
+        db=db, firebase_id=firebase_id, clearance="player"
+    )
+    if access:
+        tournaments = crud.get_tournaments(db, skip=skip, limit=limit)
+        return tournaments
+    else:
+        raise HTTPException(
+            status_code=404, detail="Permission denied, requires at least: player"
```
Comment (bakowroc) [inline backend/app/main.py:480]: Same here  `401` / `403`

File: backend/app/main.py [hunk 7]
```diff
@@ -407,10 +428,184 @@ def update_match(
...
+        db=db, firebase_id=firebase_id, clearance="player"
+    )
+    if access:
+        db_tournament = crud.get_tournament(db, tournament_id=tournament_id)
+        if db_tournament is None:
+            raise HTTPException(status_code=404, detail="Tournament not found")
+        return db_tournament
+    else:
+        raise HTTPException(
+            status_code=404, detail="Permission denied, requires at least: player"
```
Comment (bakowroc) [inline backend/app/main.py:499]: `401` / `403`. Looks like this access checking can be an utility function, its used in many places

**Pull request merged**